detect/detect.go

```diff
@@ -284,9 +284,16 @@ func (d *Detector) detectRule(fragment Fragment, rule config.Rule) []report.Find
 			continue
 		}
 
-		// check if the secret is in the allowlist
-		if rule.Allowlist.RegexAllowed(finding.Secret) ||
-			d.Config.Allowlist.RegexAllowed(finding.Secret) {
+		// check if the regexTarget is defined in the allowlist "regexes" entry
+		allowlistTarget := finding.Secret
+		switch rule.Allowlist.RegexTarget {
+		case "match":
+			allowlistTarget = finding.Match
+		case "line":
+			allowlistTarget = finding.Line
+		}
+		if rule.Allowlist.RegexAllowed(allowlistTarget) ||
+			d.Config.Allowlist.RegexAllowed(allowlistTarget) {
```

Comment:
@zricethezav wouldn't you need to set the allowlistTarget twice like: `ruleAllowListTarget` set by rule.Allowlist.RegexTarget and `globalAllowListTarget` set by d.Config.Allowlist.RegexTarget?

And then this check would become something like:

```go
		if rule.Allowlist.RegexAllowed(ruleAllowlistTarget) ||
			d.Config.Allowlist.RegexAllowed(globalAllowlistTarget) {
```

That way the global allowlist could set it's own regex target instead of the rule defining it which might result in weird results if different rules have different regex targets.

I could also be misunderstanding the code that you have already. Heh I'm just really dependent on the feature so I can upgrade past 7.6.1, so just making sure I understand the new change ^\_^

Reply:
@bplaxco fantastic catch, you are absolutely right. We do need two different targets here. Pushing some changes now